R/obic.R

```diff
@@ -460,11 +460,10 @@ obic_field_dt <- function(dt,output = 'all') {
               'A_CEC_CO','A_CA_CO_PO', 'A_MG_CO_PO', 'A_K_CO_PO',
               'A_K_CC', 'A_MG_CC', 'A_MN_CC', 'A_ZN_CC', 'A_CU_CC')
   
-  # check input
-  dt.check <- length(dt.req[dt.req %in% dt.cols]) == 29
-  
-  # check type of dt
-  checkmate::assert_true(dt.check)
+  # check presence of required columns
+  dt.check <- dt.cols[dt.cols %in% dt.req]
+  checkmate::assert_true(all(dt.req %in% dt.check & dt.check %in% dt.req),
```

Comment:
Is `all(colnames(dt) %in% dt.req),` not an more simpler option with the same result?

Reply:
dt may contain column names that are not required such as `A_C_BCS`
So, while simpler, `all(colnames(dt) %in% dt.req)` will not give the same result

However, adding such a check may be a solution to #142 if we check for allowed columns (which include \_BCS, M\_ and ID column names)
`all(colnames(dt) %in% dt.allowed)`

Reply:
Yes, but if `dt` contains not needed columns it will not give an alert, only if a required column is missing. So do we need an alert if if `dt` contains columns that are not required?

Reply:
An error is needed when columns in `dt.required` are missing, without these `obic_field()` will raise errors down the road.
I think a warning or error should be raised when NOT `all(colnames(dt) %in% dt.allowed)` as this may be an indication of misspellings, deprecated parameter names or overall messiness.

I suggest two checks:
1. `all(dt.req %in% colnames(dt))` to check that all obligatory columns are present
2. `all(colnames(dt) %in% dt.allowed)` to check if only expected colnames are present; `dt.allowed` consisting of `dt.req` and columns for management and BCS.

Reply:
But why should 2 not be allowed?